stock_packed_picking/models/stock_picking.py

```diff
@@ -0,0 +1,105 @@
+from odoo import Command, api, models
+
+
+class Picking(models.Model):
+    _inherit = "stock.picking"
+
+    @api.model
+    def _create_packed_picking(
+        self,
+        operation_type,
+        stock_move_data,
+        owner=None,
+        location=None,
+        location_dest_id=None,
+        package_name=None,
+        create_lots=False,
+        set_ready=False,
+    ):
+        """
+        Create a picking and put its product into a a package.
+
+        This is equal to the following sequence:
+        - Create a new picking
+        - Assign an owner
+        - Add products and set qty_done
+        - Mark as "Todo"
+        - Put in pack
+
+        Args:
+        operation_type (stock.picking.type): Operation type
+        stock_move_data (List of tuples): [(product_id, qty_done, serial)]
+        - (Integer) product_id: id of the product
+        - (float) qty_done: quantity done
+        - (Char, optional) serial: serial number to assign.
+        Default lot names will be used if is None or == False
+        Used only if 'create_lots==True'
+        owner (res.partner, optional): Owner of the product
+        location (stock.location, optional): Source location if differs from the
+        operation type one
+        location_dest (stock.location, optional): Destination location if differs
+        from the operation type one
+        package_name (Char, optional): Name to be assigned to the package. Default
+        name will be used if not provided.
+        set_ready (bool, optional): Try to set picking to the "Ready" state.
+
+        Returns:
+            stock.picking: Created picking
+        """
+        picking_vals = {
+            "picking_type_id": operation_type.id,
+            "owner_id": owner.id if owner else False,
+        }
+        if location:
+            picking_vals["location_id"] = location.id
+        if location_dest_id:
+            picking_vals["location_dest_id"] = location_dest_id.id
+        picking = self.env["stock.picking"].create(picking_vals)
+        picking.write(
+            {
+                "move_ids": [
+                    Command.create(
+                        {
```

Comment:
It's reasonable to move such things as value preparation into a separate function.
Because it's much easier to inherit and modify such auxiliary functions. [Example](https://github.com/odoo/odoo/blob/b90503cb25893fcdf9fbcf8e896d586a45b7552f/addons/hr_expense/models/hr_expense.py#L1256)